Rogelio_respository_implementation

RogerPerva opened the pull request:
Implementation of entities, queries, and changed the logic for some methods in order to use the H2 database

RogerPerva committed:
Creating Entities and repositories for each class.

RogerPerva committed:
Continue creating Queries, started creating a own class for player in order to understand more the code, changed the methods to entities and changed the logic for those methods so now we can use a virtual data base called h2, set names for some values in players, and continue working on it

RogerPerva committed:
Testing

RogerPerva committed:
Created more services in order to maintain clean the structure, separated the services.
-Separated responsibilities between PlayerRepository and TeamRepository for player and team operations.
-implemented the Queries for delete and edit.
- Refactored PlayerService to include deletePlayersByTeam method with transactional annotation.
-Unified the Controllers in LeagueController.
-Added the value League to TeamEntity and TeamDTO in order to improve the implementation of TeamService and TeamRepository

RogerPerva committed:
eliminate the profile from basketleague and SoccerLeague and moved into initializerData

fjpgtt commented on src/main/java/com/iwaconsolti/league/demo/service/ValidateService.java:
```diff
@@ -8,32 +9,29 @@
-            return soccerLeague.getTeamDTOS().stream().anyMatch((team -> team.getName().equalsIgnoreCase(teamName)));
+    public boolean validationTeamName(String teamName, String leagueName) {
+        if (validationLeagueName(leagueName)) {
+            log.info("Team '{}' exists in league '{}'", teamName, leagueName);
```
This log is misleading, since you havent check id the team exist in this league. that is done in the next line

fjpgtt commented on src/main/java/com/iwaconsolti/league/demo/service/ValidateService.java:
```diff
@@ -8,32 +9,29 @@
+            log.info("Team '{}' exists in league '{}'", teamName, leagueName);
+            return teamRepository.existsByName(teamName);
         }
+        log.error("Invalid league name: {}", leagueName);
```
This log is almos the sam that the one in validationLeagueName so we are going to have 2 logs almost the same

fjpgtt commented on src/main/java/com/iwaconsolti/league/demo/controller/PlayerController.java:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    @GetMapping("/{teamName}")
+    public List<PlayerEntity> getPlayersByTeam(@PathVariable String teamName) {
```
Remember to not return the same object as the entity

fjpgtt commented on src/main/java/com/iwaconsolti/league/demo/controller/PlayerController.java:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    @PostMapping
+    public ResponseEntity<PlayerEntity> createPlayer(@RequestBody PlayerEntity player) {
```
do not receive the entity as the request

fjpgtt commented on src/main/java/com/iwaconsolti/league/demo/controller/PlayerController.java:
Why you have these endpoints where no league is mention?

fjpgtt commented on src/main/java/com/iwaconsolti/league/demo/dto/MatchDTO.java:
```diff
@@ -1,13 +1,15 @@
 package com.iwaconsolti.league.demo.dto;
 
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
```
You are ot using these imports

fjpgtt commented on src/main/java/com/iwaconsolti/league/demo/dto/PlayerDTO.java:
```diff
@@ -1,19 +1,19 @@
 package com.iwaconsolti.league.demo.dto;
 
+import jakarta.persistence.*;
```
This is not used

fjpgtt commented on src/main/java/com/iwaconsolti/league/demo/dto/TeamDTO.java:
```diff
@@ -1,5 +1,6 @@
 package com.iwaconsolti.league.demo.dto;
 
+import jakarta.persistence.*;
```
Not used

fjpgtt commented on src/main/java/com/iwaconsolti/league/demo/entity/MatchEntity.java:
```diff
@@ -0,0 +1,24 @@
+@Table(name="matches")
+@Getter
+@Setter
+public class MatchEntity {
```
I dont see any attribute that indicate the score of the match

fjpgtt commented on src/main/java/com/iwaconsolti/league/demo/entity/TeamEntity.java:
```diff
@@ -0,0 +1,30 @@
+    @Column(nullable = false)
+    private String name;
+    @Column(nullable = false)
+    private int score;
```
The score should be in the match not in the team

fjpgtt commented on src/main/java/com/iwaconsolti/league/demo/entity/MatchEntity.java:
```diff
@@ -0,0 +1,24 @@
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long ID;
+
+    @ManyToOne
```
You need to indicate that these are lazy fetch

fjpgtt commented on src/main/java/com/iwaconsolti/league/demo/entity/TeamEntity.java:
```diff
@@ -0,0 +1,30 @@
+    @Column(nullable = false)
+    private int score;
+
+    @OneToMany(mappedBy = "team")
```
Make it lazy fetch

fjpgtt commented on src/main/java/com/iwaconsolti/league/demo/entity/TeamEntity.java:
```diff
@@ -0,0 +1,30 @@
+    private List<PlayerEntity> players = new ArrayList<>();
+
+
+    public TeamEntity(long id, String name, int score) {
```
Why this constructor?

fjpgtt commented on src/main/java/com/iwaconsolti/league/demo/entity/PlayerEntity.java:
```diff
@@ -0,0 +1,23 @@
+    @Column(nullable = false)
+    private String name;
+
+    @Column(name="teamName",nullable = false)
```
This should be a relationship to the TeamEntity object

fjpgtt commented on src/main/java/com/iwaconsolti/league/demo/repository/PlayerRepository.java:
```diff
@@ -0,0 +1,17 @@
+public interface PlayerRepository extends JpaRepository<PlayerEntity, Long> {
+
+    @Query("SELECT p FROM PlayerEntity p WHERE p.team = :teamName")
+    List<PlayerEntity> findByTeam(@Param("teamName") String teamName);
```
This can also be done with the method name

fjpgtt commented on src/main/java/com/iwaconsolti/league/demo/repository/PlayerRepository.java:
```diff
@@ -0,0 +1,17 @@
+    @Query("SELECT p FROM PlayerEntity p WHERE p.team = :teamName")
+    List<PlayerEntity> findByTeam(@Param("teamName") String teamName);
+
+    @Query("DELETE FROM PlayerEntity p WHERE p.team = :teamName")
```
You need to add [@]Modifying to indicate the query affects the data

RogerPerva committed:
Resolved comments made by Frank.
-Created a Enum list in order to make a list of the leagues in case we need to add more.
-Changed value in editTeam where we were receiving int instead of long and that was causing issues
-Implemented a method in for validate the league so we can use it and not duplicate code anymore
-tested with default profile and populated profile.
-Created a method to convert DTo into entity
-Tested some endpoints
-adjust correctly entities in matchEntity.
-Added score into matchEntity
-Added logs and exceptions to some methods
-Change the endpoint into DTOS.
-Added league to Team and Player in order to reuse Services
-Created a new Postmanfile

RogerPerva committed:
Added the CORS in order to connect it with the front end.

Edited the Endpoints in order the front end can read the response, changed into Maps

Created more teams in the initializer Data
